### PR TITLE
Add Colab notebooks to run text classification model pipelines

### PR DESCRIPTION
**Pull request by Narwhalprime**
<br>
This PR adds two colab notebooks:

1) Create and run a Vertex AI pipeline to train a text classification model, using a pre-built Docker image (restricted visibility, called using CustomJob) as the training component that offers many tuning options and utilizes TFHub models (`cloud_natural_language_pipeline.ipynb`).

2) Similar pipeline but with a more streamlined, publicly-available training component (`ready_to_go_text_classification_pipeline.ipynb`).
<br><br><br>

**REQUIRED:** Fill out the below checklists or remove if irrelevant
1. If you are opening a PR for `Official Notebooks` under the [notebooks/official](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/official) folder, follow this mandatory checklist:
- [ ] Use the [notebook template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb) as a starting point.
- [ ] Follow the style and grammar rules outlined in the above notebook template.
- [ ] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.
- [ ] Passes all the required automated checks. You can locally test for formatting and linting with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).
- [ ] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [ ] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/CODEOWNERS) file under the `Official Notebooks` section, pointing to the author or the author's team.
- [ ] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

<br>

2. If you are opening a PR for `Community Notebooks` under the [notebooks/community](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/community) folder:
- [x] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/community/CODEOWNERS) file under the `Community Notebooks` section, pointing to the author or the author's team.
- [X] Passes all the required formatting and linting checks. You can locally test with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).

<br>

3. If you are opening a PR for `Community Content` under the [community-content](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/community-content) folder:
- [] Make sure your main `Content Directory Name` is descriptive, informative, and includes some of the key products and attributes of your content, so that it is differentiable from other content
- [] The main content directory has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/community-content/CODEOWNERS) file under the `Community Content` section, pointing to the author or the author's team.
- [] Passes all the required formatting and linting checks. You can locally test with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).

### CODE REVIEW

File: notebooks/community/pipelines/google_cloud_pipeline_components_cloud_natural_language_pipeline.ipynb [hunk 1]
```diff
@@ -0,0 +1,813 @@
+{
```
Comment (andrewferlitsch) [inline notebooks/community/pipelines/google_cloud_pipeline_components_cloud_natural_language_pipeline.ipynb:1]: <p>per template, combine this and next cell</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1360/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (andrewferlitsch) [inline notebooks/community/pipelines/google_cloud_pipeline_components_cloud_natural_language_pipeline.ipynb:1]: <p>per template:</p><p>Missing objective section</p><p>Should be split into separate cells: Overview, Objective, Dataset, Costs</p><p> </p><p>Vertex dataset -&gt; Vertex AI Dataset</p><p>Vertex model registry -&gt; Vertex AI Model Registry</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1360/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (andrewferlitsch) [inline notebooks/community/pipelines/google_cloud_pipeline_components_cloud_natural_language_pipeline.ipynb:1]: <p>Can you describe the components here?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1360/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='36'/>

Comment (andrewferlitsch) [inline notebooks/community/pipelines/google_cloud_pipeline_components_cloud_natural_language_pipeline.ipynb:1]: <p>Describe what the pipeline does.</p><p>Remove commented out lines</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1360/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='38'/>

File: notebooks/community/pipelines/google_cloud_pipeline_components_ready_to_go_text_classification_pipeline.ipynb [hunk 2]
```diff
@@ -0,0 +1,1198 @@
+{
```
Comment (andrewferlitsch) [inline notebooks/community/pipelines/google_cloud_pipeline_components_ready_to_go_text_classification_pipeline.ipynb:1]: <p>Add H1 title to this cell</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1360/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (andrewferlitsch) [inline notebooks/community/pipelines/google_cloud_pipeline_components_ready_to_go_text_classification_pipeline.ipynb:1]: <p>describe what the pipeline does</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1360/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='51'/>

File: notebooks/community/pipelines/google_cloud_pipeline_components_cloud_natural_language_pipeline.ipynb (hunk 1 above)
Comment (Narwhalprime) [reply notebooks/community/pipelines/google_cloud_pipeline_components_cloud_natural_language_pipeline.ipynb:1]: Done.

Comment (Narwhalprime) [reply notebooks/community/pipelines/google_cloud_pipeline_components_cloud_natural_language_pipeline.ipynb:1]: Done.

Comment (Narwhalprime) [reply notebooks/community/pipelines/google_cloud_pipeline_components_cloud_natural_language_pipeline.ipynb:1]: Done.

Comment (Narwhalprime) [reply notebooks/community/pipelines/google_cloud_pipeline_components_cloud_natural_language_pipeline.ipynb:1]: Done.

File: notebooks/community/pipelines/google_cloud_pipeline_components_ready_to_go_text_classification_pipeline.ipynb (hunk 2 above)
Comment (Narwhalprime) [reply notebooks/community/pipelines/google_cloud_pipeline_components_ready_to_go_text_classification_pipeline.ipynb:1]: Done.

[the remark this answers is not in the document]
Comment (Narwhalprime) [reply notebooks/community/pipelines/google_cloud_pipeline_components_ready_to_go_text_classification_pipeline.ipynb:1]: Done.

[the remark this answers is not in the document]
Comment (Narwhalprime) [reply notebooks/community/pipelines/google_cloud_pipeline_components_ready_to_go_text_classification_pipeline.ipynb:1]: Done.

[the remark this answers is not in the document]
Comment (Narwhalprime) [reply notebooks/community/pipelines/google_cloud_pipeline_components_ready_to_go_text_classification_pipeline.ipynb:1]: Done (replaced all instance of timestamp with UUID)

[the remark this answers is not in the document]
Comment (Narwhalprime) [reply notebooks/community/pipelines/google_cloud_pipeline_components_ready_to_go_text_classification_pipeline.ipynb:1]: Removed comment.

[the remark this answers is not in the document]
Comment (Narwhalprime) [reply notebooks/community/pipelines/google_cloud_pipeline_components_ready_to_go_text_classification_pipeline.ipynb:1]: Done.

[the remark this answers is not in the document]
Comment (Narwhalprime) [reply notebooks/community/pipelines/google_cloud_pipeline_components_ready_to_go_text_classification_pipeline.ipynb:1]: Done.

Comment (Narwhalprime) [reply notebooks/community/pipelines/google_cloud_pipeline_components_ready_to_go_text_classification_pipeline.ipynb:1]: Done.

[the remark this answers is not in the document]
Comment (Narwhalprime) [reply notebooks/community/pipelines/google_cloud_pipeline_components_ready_to_go_text_classification_pipeline.ipynb:1]: Done.

File: notebooks/community/pipelines/google_cloud_pipeline_components_cloud_natural_language_pipeline.ipynb [hunk 3]
```diff
@@ -0,0 +1,826 @@
...
```
Comment (andrewferlitsch) [inline notebooks/community/pipelines/google_cloud_pipeline_components_cloud_natural_language_pipeline.ipynb:1]: <p>use the template version</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1360/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (andrewferlitsch) [inline notebooks/community/pipelines/google_cloud_pipeline_components_cloud_natural_language_pipeline.ipynb:1]: <p>why is CLASS_NAMES empty?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1360/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Comment (andrewferlitsch) [inline notebooks/community/pipelines/google_cloud_pipeline_components_cloud_natural_language_pipeline.ipynb:1]: <p>notebook seems incomplete. For example, may show that the model was trained and the. evaluation results</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1360/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='44'/>

Comment (Narwhalprime) [reply notebooks/community/pipelines/google_cloud_pipeline_components_cloud_natural_language_pipeline.ipynb:1]: <p>Users are expected to fill in the class names of their dataset here. Unfortunately there is no Colab form for string lists.</p>

Comment (Narwhalprime) [reply notebooks/community/pipelines/google_cloud_pipeline_components_cloud_natural_language_pipeline.ipynb:1]: <p>Added cell to check metrics and instructions on viewing model in Vertex UI.</p>

**Review by andrewferlitsch — APPROVED** [review]

**Pull request merged**